[CS-3380] Claim reward transaction screen

oidoug opened the pull request:
<!-- This is a pull request. Please make sure any applicable bullet points have been covered. -->

### Description

This PR adds a new reward claim confirmation sheet, that shows data from `mainPoolTokenInfo` and uses the claim mutation callback to pop the stack back to the main rewards screen on success.

From the linear ticket, we are still missing proper Gas Estimation. We will need a new ticket for adding that once it's ready in the SDK.

- [x] Completes #(CS-3380)

### Checklist

- [x] All UI changes have been tested on a small device

### Screenshots

https://user-images.githubusercontent.com/129619/160481905-8c80d93d-f1ae-49f6-a1d7-133adda4c7fe.mp4

Commit:
Updates RewardsCenterScreen navbar title

Commit:
WIP new Rewards Claim Confirmation.

Commit:
WIP Adds custom RewardsClaimDisplay

Commit:
Adds Claim Confirmation Sheet;
Pulls info from mainPoolTokenInfo;
Changes shouldGoBack to a counter popStackNavigation, so both register and claim can use the same mutation alert handler.

Commit:
Fixes lint issue

danibonilha commented on cardstack/src/screens/RewardsCenterScreen/RewardsClaimSheet/RewardsClaimSheet.tsx:
```diff
@@ -0,0 +1,33 @@
+  SafeAreaView,
+} from '@cardstack/components';
+
+interface RouteParams {
+  data: TransactionConfirmationData;
+  onConfirm: () => Promise<void>;
+  onCancel: () => void;
+}
```
This can be a shared type, since we are using on register too

danibonilha commented on cardstack/src/screens/RewardsCenterScreen/useRewardsCenterScreen.ts:
```diff
@@ -227,16 +229,33 @@ export const useRewardsCenterScreen = () => {
+              safeAddress: rewardSafeForProgram?.address || '',
+            });
+          },
+          onCancel: () => {
+            goBack();
+          },
```
```suggestion
          onCancel:  goBack
```

danibonilha commented on cardstack/src/components/TransactionConfirmationSheet/displays/components/sections/NetClaimAmountSection.tsx:
```diff
@@ -0,0 +1,38 @@
+    <Container>
+      <SectionHeaderText>{headerText}</SectionHeaderText>
+      <Container marginLeft={15} marginTop={2}>
+        <Text size="xxs">{strings.rewards.claim.breakdownSection.reward}</Text>
```
we should either make this component more generic meaning there's no reward references, and it can be used on different  displays OR specify that its RewardNetClaimSection so we don't confuse and try to use on a diff place

oidoug replied:
I'm not sure this "net value" UI pattern will stick, so I prefer to rename it and keep it contextual to rewards.

danibonilha commented on cardstack/src/components/TransactionConfirmationSheet/displays/components/sections/NetClaimAmountSection.tsx:
```diff
@@ -0,0 +1,38 @@
+
+        <Text size="xxs">{strings.rewards.claim.breakdownSection.estNet}</Text>
+        <Text size="body" weight="bold" paddingBottom={3}>
+          {Number(balance?.amount) - (spendAmount || 0)} {token?.symbol}
```
Let's extract this to a variable, just to clean up,  also we should rename `spendAmount` to `gasFee`, `estGasFee` and avoid spendAmount, we were using `spendAmount` bc the way choosePrepaidCard was structured when spend was relevant, but it's not anymore.

oidoug replied:
Got it. I've used spendAmount bc theres a lot of similar uses with that name, like in `PayThisAmountSection`, we might want to refactor those then.

I'll be using `estGasFee` on these new sections then, seems more explicative.

Commit:
Renames spendAmount to estGasFee in Rewards Register and Claim data types;
Renames NetClaimAmountSection to RewaresNetAmountSection;
Extracts reused Route Params for Rewards Sections.

danibonilha reviewed (approved):
LGTM

danibonilha commented on cardstack/src/components/TransactionConfirmationSheet/displays/RewardsRegisterDisplay.tsx:
```diff
@@ -37,7 +37,7 @@ export const RewardsRegisterDisplay = ({
     <PayThisAmountSection
       headerText={strings.rewards.cost.title}
-      spendAmount={spendAmount}
+      spendAmount={estGasFee}
```
Although for this case I'm not sure it should be replaced, since `PayThisAmountSection` uses spend to convert to a native value.

Commit:
Reverts estGasFee to spendAmount for RewardsRegisterData Type
